plugins/hive-plugin/src/main/java/edu/berkeley/ground/plugins/hive/GroundDatabase.java

```diff
@@ -43,311 +45,171 @@
 import java.util.Map;
 
 public class GroundDatabase {
-  static final private Logger LOG = LoggerFactory.getLogger(GroundDatabase.class.getName());
+    private static final String DATABASE = "database";
 
-  static final String DATABASE_NODE = "_DATABASE";
+    static final private Logger LOG = LoggerFactory.getLogger(GroundDatabase.class.getName());
 
-  static final String DATABASE_TABLE_EDGE = "_DATABASE_TABLE";
+    static final String DATABASE_NODE = "_DATABASE";
 
-  private static final List<String> EMPTY_PARENT_LIST = new ArrayList<String>();
+    static final String DATABASE_TABLE_EDGE = "_DATABASE_TABLE";
 
-  private GroundReadWrite groundReadWrite = null;
-  private GroundTable groundTable = null;
+    private static final String DB_STATE = "_DATABASE_STATE";
 
-  GroundDatabase(GroundReadWrite ground) {
-    groundReadWrite = ground;
-    groundTable = new GroundTable(ground);
-  }
+    private static final String DATABASE_PARENT_NODE = "_DATABASE_PARENT_NODE";
 
-  Node getNode(String dbName) throws GroundException {
-    try {
-      LOG.debug("Fetching database node: " + dbName);
-      return groundReadWrite.getNodeFactory().retrieveFromDatabase(dbName);
-    } catch (GroundException ge1) {
-      LOG.debug("Not found - Creating databsae node: {}", dbName);
+    private GroundReadWrite groundReadWrite = null;
+    private GroundTable groundTable = null;
+    private NodeVersion metaDatabaseNodeVersion =null;
 
-      Node node = groundReadWrite.getNodeFactory().create(dbName, new HashMap<>());
-      Structure nodeStruct = groundReadWrite.getStructureFactory().create(node.getName(), new HashMap<>());
-      LOG.debug("node structure created {}", nodeStruct);
-      return node;
+    GroundDatabase(GroundReadWrite groundReadWrite) {
```

Comment:
Well it is scoped. These are internal methods - visible at package level and for tests. I can change